modin/experimental/cloud/rpyc_proxy.py

```diff
@@ -0,0 +1,327 @@
+                    pass  # not a get/set data descriptor, go next
+                else:
+                    return oget(self, type(self))
+            # type(self).name is not a get/set data descriptor
+            try:
+                return dct[name]
+            except KeyError:
+                # instance doesn't have an attribute
+                if has_cls_attr:
+                    # type(self) has this attribute, but it's not a get/set descriptor
+                    if oget:
+                        # this attribute is a get data descriptor
+                        return oget(self, type(self))
+                    return cls_attr  # not a data descriptor whatsoever
+
+            # this instance/class does not have this attribute, pass it through to remote end
+            return getattr(dct["__remote_end__"], name)
+
+        if override.__setattr__ == object.__setattr__:
+            # no custom attribute setting, define our own relaying to remote end
+            def __setattr__(self, name, value):
+                if name not in _PROXY_LOCAL_ATTRS:
+                    setattr(self.__remote_end__, name, value)
+                else:
+                    object.__setattr__(self, name, value)
+
+        if override.__delattr__ == object.__delattr__:
+            # no custom __delattr__, define our own
+            def __delattr__(self, name):
+                if name not in _PROXY_LOCAL_ATTRS:
+                    delattr(self.__remote_end__, name)
+
+    return Wrapper
+
+
+def _deliveringWrapper(origin_cls, methods=(), mixin=None, target_name=None):
+    conn = get_connection()
+    remote_cls = getattr(conn.modules[origin_cls.__module__], origin_cls.__name__)
+
+    if mixin is None:
+
+        class DeliveringMixin:
+            pass
+
+        mixin = DeliveringMixin
+
+    for method in methods:
+
+        def wrapper(self, *args, __remote_conn__=conn, __method_name__=method, **kw):
+            args = tuple(__remote_conn__.deliver(x) for x in args)
+            kw = {k: __remote_conn__.deliver(v) for k, v in kw.items()}
+            return getattr(self.__remote_end__, __method_name__)(*args, **kw)
+
+        wrapper.__name__ = method
+        setattr(mixin, method, wrapper)
+    return make_proxy_cls(
+        remote_cls, origin_cls, mixin, target_name or origin_cls.__name__
+    )
+
+
+def _prepare_loc_mixin():
+    from modin.pandas.indexing import _LocIndexer, _iLocIndexer
+
+    DeliveringLocIndexer = _deliveringWrapper(
+        _LocIndexer, ["__getitem__", "__setitem__"]
+    )
+    DeliveringILocIndexer = _deliveringWrapper(
+        _iLocIndexer, ["__getitem__", "__setitem__"]
+    )
+
+    class DeliveringMixin:
+        @property
+        def loc(self):
+            return DeliveringLocIndexer(self.__remote_end__)
+
+        @property
+        def iloc(self):
+            return DeliveringILocIndexer(self.__remote_end__)
+
+    return DeliveringMixin
+
+
+def make_dataframe_wrapper(DataFrame):
```

Comment:
Same here about comment for why this is needed.

Reply:
For all of these wrappers.

Reply:
added docstrings